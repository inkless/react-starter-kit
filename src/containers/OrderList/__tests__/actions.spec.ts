import configureStore from 'redux-mock-store';

import {
  delivering,
  pending,
  delivered,
} from '@src/components/OrderList/__tests__/__fixture__/orders';
import {
  updatePendingOrders,
  startDeliveringOrder,
  orderDelivered,
  fetchDestinationAction,
} from '../actions';

const mockStore = configureStore();
const store = mockStore();

describe('OrderList Actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('should dispatch updatePendingOrders', () => {
    store.dispatch(updatePendingOrders(pending));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch startDeliveringOrder', () => {
    store.dispatch(startDeliveringOrder(delivering[0]));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch orderDelivered', () => {
    store.dispatch(orderDelivered(delivered[0]));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch fetchDestinationAction', () => {
    store.dispatch(fetchDestinationAction.request());
    expect(store.getActions()).toMatchSnapshot();
  });
});
