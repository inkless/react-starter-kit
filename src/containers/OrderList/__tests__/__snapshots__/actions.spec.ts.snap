// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList Actions should dispatch fetchDestinationAction 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "@@OrderList/FETCH_DEST_REQUEST",
  },
]
`;

exports[`OrderList Actions should dispatch orderDelivered 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": Object {
      "coordinates": Array [
        -130,
        37,
      ],
      "destination": "267 N Canon Dr, Beverly Hills, CA 90210",
      "distance": 1,
      "endTime": 2019-04-20T05:11:15.265Z,
      "id": "UWkU5v",
      "name": "Mint Chocolate Ice Cream",
      "route": Object {
        "geometry": Object {
          "coordinates": Array [
            Array [
              -118.46742,
              34.00534,
            ],
            Array [
              -118.47105,
              34.00321,
            ],
          ],
          "type": "LineString",
        },
        "properties": Object {},
        "type": "Feature",
      },
      "speed": 0.0075,
      "startTime": 2019-04-20T05:00:15.265Z,
    },
    "type": "@@OrderList/ORDER_DELIVERED",
  },
]
`;

exports[`OrderList Actions should dispatch startDeliveringOrder 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": Object {
      "coordinates": Array [
        -130,
        37,
      ],
      "destination": "400 World Way, Los Angeles, CA 90045",
      "distance": 1,
      "id": "ZEUn6r",
      "name": "Chunky Monkey",
      "route": Object {
        "geometry": Object {
          "coordinates": Array [
            Array [
              -118.46742,
              34.00534,
            ],
            Array [
              -118.47105,
              34.00321,
            ],
          ],
          "type": "LineString",
        },
        "properties": Object {},
        "type": "Feature",
      },
      "speed": 0.0075,
      "startTime": 2019-04-20T05:01:15.265Z,
    },
    "type": "@@OrderList/START_DELIVERING_ORDER",
  },
]
`;

exports[`OrderList Actions should dispatch updatePendingOrders 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": Array [
      Object {
        "destination": "185 Culver Blvd, Playa Del Rey, CA 90293",
        "dispatchTime": 119,
        "id": "9H4whD",
        "name": "Banana Split",
      },
      Object {
        "destination": "13455 Maxella Ave, Marina Del Rey, CA 90292",
        "dispatchTime": 98,
        "id": "EUtSqE",
        "name": "McFlury",
      },
      Object {
        "destination": "4720 Admiralty Way, Marina Del Rey, CA 90292",
        "dispatchTime": 35,
        "id": "uoierx",
        "name": "Acai Bowl",
      },
      Object {
        "destination": "220 Rose Ave, Venice, CA 90291",
        "dispatchTime": 105,
        "id": "oP6QHd",
        "name": "Yogurt",
      },
      Object {
        "destination": "2323 Wilshire Blvd, Santa Monica, CA 90403",
        "dispatchTime": 73,
        "id": "aS4EFD",
        "name": "Chocolate Gelato",
      },
      Object {
        "destination": "1630 Ocean Park Blvd, Santa Monica, CA 90405",
        "dispatchTime": 13,
        "id": "4WVKZu",
        "name": "Cobb Salad",
      },
      Object {
        "destination": "1416 4th St, Santa Monica, CA 90401",
        "dispatchTime": 3,
        "id": "5sSNvC",
        "name": "Cottage Cheese",
      },
      Object {
        "destination": "8115 W Manchester Ave, Playa Del Rey, CA 90293",
        "dispatchTime": 119,
        "id": "Fv8AdV",
        "name": "Coke",
      },
      Object {
        "destination": "2055 Sawtelle Blvd, Los Angeles, CA 90025",
        "dispatchTime": 49,
        "id": "WFG4B3",
        "name": "Snow Cone",
      },
      Object {
        "destination": "2906 Main St, Santa Monica, CA 90405",
        "dispatchTime": 94,
        "id": "woqBZv",
        "name": "Pad See Ew",
      },
    ],
    "type": "@@OrderList/UPDATE_PENDING_ORDERS",
  },
]
`;
